From f17e8092077d79d47c7790b6025746335083f7d7 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:49 -0400
Subject: [ethernet] qlcnic: get rid of SET_ETHTOOL_OPS

Message-id: <1408626422-3941-126-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88627
O-Subject: [PATCH RHEL 7.1 125/138] net: get rid of SET_ETHTOOL_OPS
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Wilfried Klaebe <w-lkml@lebenslange-mailadresse.de>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 7ad24ea4bf620a32631d7b3069c3e30c078b0c3e

net: get rid of SET_ETHTOOL_OPS

Dave Miller mentioned he'd like to see SET_ETHTOOL_OPS gone.
This does that.

Mostly done via coccinelle script:
@@
struct ethtool_ops *ops;
struct net_device *dev;
@@
-       SET_ETHTOOL_OPS(dev, ops);
+       dev->ethtool_ops = ops;

Compile tested only, but I'd seriously wonder if this broke anything.

Suggested-by: Dave Miller <davem@davemloft.net>
Signed-off-by: Wilfried Klaebe <w-lkml@lebenslange-mailadresse.de>
Acked-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index 40fd447..85b1911 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2262,10 +2262,8 @@ qlcnic_setup_netdev(struct qlcnic_adapter *adapter, struct net_device *netdev,
 
  qlcnic_change_mtu(netdev, netdev->mtu);
 
- if (qlcnic_sriov_vf_check(adapter))
-  SET_ETHTOOL_OPS(netdev, &qlcnic_sriov_vf_ethtool_ops);
- else
-  SET_ETHTOOL_OPS(netdev, &qlcnic_ethtool_ops);
+ netdev->ethtool_ops = (qlcnic_sriov_vf_check(adapter)) ?
+  &qlcnic_sriov_vf_ethtool_ops : &qlcnic_ethtool_ops;
 
  netdev->features |= (NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_RXCSUM |
         NETIF_F_IPV6_CSUM | NETIF_F_GRO |
@@ -2679,7 +2677,7 @@ err_out_disable_pdev:
 err_out_maintenance_mode:
  set_bit(__QLCNIC_MAINTENANCE_MODE, &adapter->state);
  netdev->netdev_ops = &qlcnic_netdev_failed_ops;
- SET_ETHTOOL_OPS(netdev, &qlcnic_ethtool_failed_ops);
+ netdev->ethtool_ops = &qlcnic_ethtool_failed_ops;
  ahw->port_type = QLCNIC_XGBE;
 
  if (qlcnic_83xx_check(adapter))
-- 
1.7.1